packages/passport-crypto/src/passportCrypto.ts

```diff
@@ -32,19 +32,29 @@ export class PCDCrypto {
     return utils.arrayBufferToHexString(arrayBuffer);
   }
 
+  public generateSalt(
+    length: number = this.sodium.crypto_pwhash_SALTBYTES
+  ): HexString {
+    if (length < this.sodium.crypto_pwhash_SALTBYTES) {
+      throw Error(
+        `Salt must be at least ${this.sodium.crypto_pwhash_SALTBYTES}`
+      );
+    }
+    const buffer = this.sodium.randombytes_buf(length);
+    return utils.arrayBufferToHexString(buffer);
+  }
+
   public argon2(
     password: Utf8String,
     salt: HexString,
-    iterations: number,
-    bytes: number,
     length: number
   ): HexString {
     const result = this.sodium.crypto_pwhash(
       length,
       utils.stringToArrayBuffer(password),
       utils.hexStringToArrayBuffer(salt),
-      iterations,
-      bytes,
+      this.sodium.crypto_pwhash_OPSLIMIT_INTERACTIVE,
+      this.sodium.crypto_pwhash_MEMLIMIT_INTERACTIVE,
```

Comment:
what are these constants and what do they mean?

Reply:
added documentation and linked source